### PR TITLE
Update NES30 buttonmap for latest firmware

### PR DESCRIPTION
**Pull request by EvanPurkhiser**
Adds support for the NES30 in normal "Joystick" mode, which is different from the currently supported NES30 in "PC Joystick" mode. [See the v2.52 firmware notes](http://www.nes30.com/firmware.html).

---

#### Old PR description:

It appears in the latest firmware of the NES30 the device reports **16** buttons, not **22**. This allows XBMC to correctly identify it.

This also corrects the name of the device as it does not include `Joystick` in bluetooth mode.

I've never seen my NES30 report itself with the suffix `Joystick` in bluetooth mode. I am on Arch Linux however, maybe it reports differently on Windows and OSX. It's quite possible this device should have an alias added without the Joystick suffix. Pinging @N3MIS15 for information about his setup.

### CODE REVIEW

**Comment by N3MIS15** [comment]
The original button map is created when holding r+start to put the controller in joystick mode. Starting it in normal mode (previously, unsure about current firmware) detected the dpad as keyboard and the rest of the buttons as a joypad (2 separate devices). Joystick mode solved this issue.

**Comment by N3MIS15** [comment]
FWIW, I remember it working fine in the menus, but not game mode.

**Comment by EvanPurkhiser** [comment]
Thanks for the quick response @N3MIS15!

I think there's a little bit of confusion of what "Joystick mode" is. Looking at [the manual](http://8bitdo.com/nes30pro/Manual.html) it actually says joystick mode is when you power it up without holding any buttons. *However*, there is a **PC Joystick** mode, which is as you describe.

> detected the dpad as keyboard and the rest of the buttons as a joypad (2 separate devices). Joystick mode solved this issue.

I had this problem (which actually worked fine for awhile, since as you said, the keyboard works in the menus. I actually don't use game mode at all, since it seems this addon is now required to use SDL joystick inputs on Kodi.

What I **can tell you now** though, is that on the latest firmware the d-pad does *not* send keyboard scancodes, it appears to now be a single device. So in that case, it seems this PR may be valid. Let me take a look at what it reports when connecting in "PC Joystick" mode now.

**Comment by EvanPurkhiser** [comment]
> 16 and not 26 buttons

Whoops! Yes, that's correct, that is a typo. I'll split this out into a separate file so we can support both :-)

**Comment by EvanPurkhiser** [comment]
Actually @N3MIS15 I'm not sure anymore. I just realized I was looking at the NES30 Pro manual, which shows the <kbd>start</kbd>+<kbd>r</kbd> combo, but [the manual for the NES30 (non pro)](http://www.nes30.com/manual.html) doesn't even show a mode that can be activated with that boot combo.

Looking at the (firmware updates page](http://www.nes30.com/firmware.html) it looks like the mode you speak of **was added** in version 2.5.2:

> 2.Added PC Joystick mode(power up START+R and connect to PC). D-Pad works as Joystick D-Pad, not a keyboard arrow key.

There's also this somewhat cryptic note

> 1.Modified Mode 2 (START+B+R) secondly key code. D-Pad change from arrow key to normally letter, in order work with emulator better.

Either way.. Booting my controller with firmware v2.70 using just <kbd>start</kbd> correctly sends joystick d-pad events.

**In conclusion** I think this is safe to merge :-)

**Comment by garbear** [comment]
Looks good now, thanks.
